WALRUS/diag_WALRUS.py

```diff
@@ -0,0 +1,211 @@
+"""Diagnostic for WALRUS based on diag_shapeselect.py."""
+import logging
+import os
+from copy import deepcopy
+
+import fiona
+import pandas as pd
+import iris
+import numpy as np
+from netCDF4 import Dataset, num2date
+from shapely.geometry import MultiPoint, shape
+from shapely.ops import nearest_points
+
+from esmvaltool.diag_scripts.shared import (run_diagnostic, ProvenanceLogger,
+                                            get_diagnostic_filename)
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+#TODO check merging the Q
+def get_provenance_record(cfg, basename, caption, extension):
+    """Create a provenance record describing the diagnostic data and plot."""
+    record = {
+        'caption': caption,
+        'statistics': ['other'],
+        'domains': ['global'],
+        'authors': ['berg_pe'],
+        'references': ['acknow_project'],
+    }
+    diagnostic_file = get_diagnostic_filename(basename, cfg, extension)
+    with ProvenanceLogger(cfg) as provenance_logger:
+        provenance_logger.log(diagnostic_file, record)
+
+
+def shapeselect(cfg, cube):
```

Comment:
You don't need to copy the shapeselect diagnostic into this one. It's possible to just call it from the recipe.